Source/WebCore/Modules/mediasession/MediaSession.h

```diff
@@ -156,11 +159,28 @@ class MediaSession : public RefCounted<MediaSession>, public ActiveDOMObject, pu
 #if ENABLE(MEDIA_SESSION_PLAYLIST)
     Vector<Ref<MediaMetadata>> m_playlist;
 #endif
+    mutable Lock m_actionHandlersLock;
 };
 
 String convertEnumerationToString(MediaSessionPlaybackState);
 String convertEnumerationToString(MediaSessionAction);
 
+inline bool MediaSession::hasActiveActionHandlers() const
+{
+    Locker lock { m_actionHandlersLock };
+    return !m_actionHandlers.isEmpty();
+}
+
+template <typename Visitor>
```

Comment:
A template function in the cpp even though the function is used from outside the cpp? I am a bit surprised this builds without explicit template instantiations. Maybe you're getting lucky because of unified builds?

Reply:
I think you'd need explicit template instantiations for  `JSC::AbstractSlotVisitor` & `JSC::SlotVisitor`.

Reply:
I'm pretty sure this is the .h file. 

I'd be happy to use explicit instantiations and move it to the cpp though if you want.

Reply:
Oh! Got confused by the diff format. You can keep as-is then. Don't know why I thought this was the cpp file :)

Reply:
I did originally try putting this in the cpp file without explicit instantiations so I got concerned I pushed a bad commit! I'll leave it alone then.